### ISSUE TITLE
HTML Entities are randomly deleted

### ISSUE THREAD

**Issue by rpopov** [typora/typora-issues#5499]
- [Yes] Searched existing issues to avoid creating duplicates.

- [Yes] Confirmed that it can be reproduced in built-in themes without customized css.

- [Yes] Searched http://support.typora.io/

----

**Describe the bug**
The HTML elements get randomly deleted (on save?). After a long period of editing and saving a long text (111.3K) with many (2490) HTML entities, I frequently realize that some of the HTML elements are deleted automatically. Expressions like (&exist;i&in;N) loose their &exist; or &in; entity. The deletion happens when concatenated to regular characters like i or N in the example. The effect could be related to the mass use of such entities, to the size of the text, or to browsing up / down the pages. Sometimes I have to go several pages up/down and go back quickly, not sure if their rendering is complete. 

Examples:
* These happened on successive lines
   *   on the same line (&exist; i &in; Id) saved as  (&exist; i  Id) and (&exist;i &in; Id saved as (i &in; Id)
   *   on the same line (&forall;c &in; C....) saved as (c  C....) and (&exist;i &in; Id saved as (i &in; Id)
* 

**Expected behavior**
The HTML entities are kept unless explicitly deleted.

**Screenshots / Screencasts**
![Untitled](https://user-images.githubusercontent.com/7579924/200193095-0b10fd90-ab41-4702-9b7d-a196e524ff1e.png)

**Sample Markdown File**
This is a diff between the detected (-) and fixed (+) lines of the .md
```
- (&exist; i  Id)(&not;!s(i) & ![s]c(i)) &or; (i &in; Id)(a &in; Attribute)(&not;!s(i,a) & ![s]c(i,a))
+ (&exist; i &in; Id)(&not;!s(i) & ![s]c(i)) &or; (&exist;i &in; Id)(a &in; Attribute)(&not;!s(i,a) & ![s]c(i,a))
 
- (c  Change<sub>0</sub>)(&forall; s &in; State)((i &in; Id)(!s(i) & ![s]c(i)) &or; (&exist; i  Id)(&exist; a &in; Attribute)(&not;!s(i,a) & ![s]c(i,a))).
+ (&forall;c &in; Change<sub>0</sub>)(&forall; s &in; State)((i &in; Id)(!s(i) & ![s]c(i)) &or; (&exist; i  Id)(&exist; a &in; Attribute)(&not;!s(i,a) & ![s]c(i,a))).
```

**Desktop (please complete the following information):**
 - OS: Oracle Linux 8.6
Linux .... 5.4.17-2136.312.3.4.el8uek.x86_64 # 2 SMP Wed Oct 19 17:42:16 PDT 2022 x86_64 x86_64 x86_64 GNU/Linux

**Typora Version** 1.4.8

**Additional context**
When surrounded with spaces the HTML entities are less likely to be deleted randomly. The regular characters around the HTML entities are not deleted.

**Comment by abnerlee** [comment]
Did you find any pattern that may trigger this?

**Comment by rpopov** [comment]
I have not confirmed a pattern here. Some initial ideas:
* I used to scroll up/down quickly and frequently, not waiting to render the whole contents
* I saw (more) frequently disappearing entities when they are joined - no space or other characters between them
* It seems restricted to 1-2 screen pages at a time - the entities do not disappear from the whole document, specifically from pages I have not read recently.
In general, the entities disappear in parts of the text that were not presently edited - I focus my editing in some place, save frequently and later reopening the document I see entities have disappeared in places out of my focus, but I had read/scroll through several times.

**Comment by rpopov** [comment]
Just noted that searching for an HTLM entity deletes it from the text. Seems related to the highlighting of the occurrences.
Now
**&asymp;** was replaced with ****

**Comment by rpopov** [comment]
No other cases of disappearing entities were found, except when searching for them. I noted also that searching for **epsilon** text also deleted the **\&epislon\;** entities.

**Comment by rpopov** [comment]
Just noticed that when typing in the Search dialog a text that is part of an HTML entity, this also deletes (some) of the entities. For example:
* I have several &delta; (\&delta\;) entities and I typed "del". While typing, the search dialog started a background search, it found the occurrences of "del" in "delta" on the current page / visible part of the text, did not show it, but the &delta; characters disappeared.

The disappeared entities were not treated as changes in the text, so there was no indication in the window's title bar the text was modified. Saving did not actually update the file on the disk. 
I added some spaces and saved - saved were only the spaces.
I changed some letters and saved the text - saved were the changed letters AND the disappeared entities were deleted.

* I have many &rarr; (\&rarr\;) and &harr; (\&harr\;) entities. When started searching for "arr", Typora highlighted all of them on the page (well, the highlight was shown at some offset right, probably related to the length of the entity as a text), and they did not disappear on the page.
Thus I think the issue is related to the class of the entities - ones that represent letters disappear, but those entities that represent something else (non-letter) are kept.

Thus the combination of search in entities, background search while typing in the Search dialog, highlighting the findings, not indicating the text was changed, and saving only when non-whitespace chars are changed makes the deletion of the entities hard to detect and track.
